File: content/snippets/python/problems_word_count.py
#!/usr/bin/env python3
from collections import Counter
import re
import sys

def word_count(path='data.txt'):
    with open(path, 'r', encoding='utf-8') as fh:
        text = fh.read().lower()
    words = re.findall(r"\b\w+\b", text)
    return Counter(words)

def main():
    path = sys.argv[1] if len(sys.argv) > 1 else 'data.txt'
    cnt = word_count(path)
    for w, c in cnt.most_common():
        print(f"{w}: {c}")

if __name__ == '__main__':
    main()

File: content/snippets/typescript/problems_word_count.ts
#!/usr/bin/env ts-node
import { promises as fs } from 'fs';

async function main() {
    const path = process.argv[2] || 'data.txt';
    const text = await fs.readFile(path, 'utf8');
    const words = text.toLowerCase().match(/\b\w+\b/g) || [];
    const m = new Map<string, number>();
    for (const w of words) m.set(w, (m.get(w) || 0) + 1);
    const arr = Array.from(m.entries()).sort((a,b) => b[1] - a[1]);
    for (const [k,v] of arr) console.log(`${k}: ${v}`);
}

main().catch(e => { console.error('Error:', e); process.exit(1); });
